core/src/main/java/org/apache/accumulo/core/file/blockfile/cache/LruBlockCache.java

```diff
@@ -143,6 +144,9 @@
   /** Overhead of the structure itself */
   private long overhead;
 
+  /** Locks used when loading data not in the cache. Size is prime number. */
+  private Lock[] loadLocks = new Lock[5003];
```

Comment:
Why an array of load locks?  Can we not create a map of name to loadlock from which the loadlock gets cleaned up (or returned to a pool) once the block is cached?  Perhaps this simpler method is better than the overhead I am suggesting but I want to make sure it was considered.

Reply:
I thought about it, I suspect this is simpler and possibly faster.   For the map approach  would need to do the following :

 * Synchronization when creating a non-existent lock to ensure two threads do not create different locks for the same key.
 * Reference count locks and remove them when not needed.  Need to sync with create lock code when removing a lock that has a ref count of zero.

That reminds me that conditional mutations take this map approach when locking rows.  Maybe that code could be improved and simplified.